In [21]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

%matplotlib inline

# Loading the housing price dataset (assuming the file name is "housing_price.csv")
df = pd.read_csv("/Users/sebastianfrowein/Desktop/Bewerbungen 2024/df_demo_merged_concat2")

/var/folders/wl/6wzqzn0n53nfpnx2t2hv7rxm0000gn/T/ipykernel_88939/1139391674.py:9: DtypeWarning: Columns (3) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("/Users/sebastianfrowein/Desktop/Bewerbungen 2024/df_demo_merged_concat2")


In [23]:
from scipy.stats import ttest_ind

if len(completion_male) == 0 or len(completion_female) == 0:
    print("Insufficient data for one or both genders.")
else:
    t_stat, p_value = ttest_ind(completion_male, completion_female, equal_var=False)

    print(f"T-statistic: {t_stat}")
    print(f"P-value: {p_value}")

    alpha = 0.05
    if p_value < alpha:
        print("Reject the null hypothesis: There is a statistically significant difference between the completion rates of males and females.")
    else:
        print("Fail to reject the null hypothesis: There is no statistically significant difference between the completion rates of males and females.")


T-statistic: 4.6158133077207255
P-value: 3.920397979906506e-06
Reject the null hypothesis: There is a statistically significant difference between the completion rates of males and females.


## Analysis and Conclusion## 
P-value Analysis: The p-value  is much smaller than the significance level (0.05). This indicates that the observed difference in completion rates between males and females is highly unlikely to have occurred by random chance.

T-statistic Analysis: the difference between the male and female completion rates is 4.6158 standard deviations away from the null hypothesis.

###Conclusion###
Reject the Null Hypothesis: Since the p-value is less than the significance level (0.05), we reject the null hypothesis. This means there is statistically significant evidence to suggest that there is a difference in the completion rates between males and females.
Why the Difference is Significant
Low P-value: The extremely low p-value (3.92e-06) indicates that the probability of observing such a difference by random chance is very low if the null hypothesis were true.
High T-statistic: A high t-statistic further supports that the difference between the groups is not due to random variation.